Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -32,44 +33,131 @@ internal class WindowsConsole {
 
 	IntPtr _inputHandle, _outputHandle;
 	IntPtr _screenBuffer;
-	readonly uint _originalConsoleMode;
+	readonly uint _originalInputConsoleMode;
+	readonly uint _originalOutputConsoleMode;
+	CONSOLE_SCREEN_BUFFER_INFOEX _originalScreenBufferInfo;
 	CursorVisibility? _initialCursorVisibility = null;
 	CursorVisibility? _currentCursorVisibility = null;
 	CursorVisibility? _pendingCursorVisibility = null;
 	readonly StringBuilder _stringBuilder = new StringBuilder (256 * 1024);
+	bool _isWindowsTerminal;
 
-	public WindowsConsole ()
+	public WindowsConsole (bool isWindowsTerminal)
 	{
 		_inputHandle = GetStdHandle (STD_INPUT_HANDLE);
 		_outputHandle = GetStdHandle (STD_OUTPUT_HANDLE);
-		_originalConsoleMode = ConsoleMode;
-		var newConsoleMode = _originalConsoleMode;
-		newConsoleMode |= (uint)(ConsoleModes.EnableMouseInput | ConsoleModes.EnableExtendedFlags);
-		newConsoleMode &= ~(uint)ConsoleModes.EnableQuickEditMode;
-		newConsoleMode &= ~(uint)ConsoleModes.EnableProcessedInput;
-		ConsoleMode = newConsoleMode;
+
+		_isWindowsTerminal = isWindowsTerminal;
+		if (!ConsoleDriver.RunningUnitTests && !_isWindowsTerminal) {
+			_originalScreenBufferInfo = new CONSOLE_SCREEN_BUFFER_INFOEX ();
+			_originalScreenBufferInfo.cbSize = (uint)Marshal.SizeOf (_originalScreenBufferInfo);
+			if (!GetConsoleScreenBufferInfoEx (_outputHandle, ref _originalScreenBufferInfo)) {
+				var err = Marshal.GetLastWin32Error ();
+				if (err != 0) {
+					throw new System.ComponentModel.Win32Exception (err);
+				}
+			}
+		}
+
+		_originalInputConsoleMode = InputConsoleMode;
+		var newInConsoleMode = _originalInputConsoleMode;
+		newInConsoleMode |= (uint)(ConsoleModes.EnableMouseInput | ConsoleModes.EnableExtendedFlags);
+		newInConsoleMode &= ~(uint)ConsoleModes.EnableQuickEditMode;
+		newInConsoleMode &= ~(uint)ConsoleModes.EnableProcessedInput;
+		InputConsoleMode = newInConsoleMode;
+
+		_originalOutputConsoleMode = OutputConsoleMode;
+		var newOutConsoleMode = _originalOutputConsoleMode;
+		if (!ConsoleDriver.RunningUnitTests && (newOutConsoleMode & (uint)(ConsoleModes.EnableVirtualTerminalProcessing | ConsoleModes.DisableNewlineAutoReturn))
+			< (uint)ConsoleModes.DisableNewlineAutoReturn) {
+
+			newOutConsoleMode |= (uint)(ConsoleModes.EnableVirtualTerminalProcessing | ConsoleModes.DisableNewlineAutoReturn);
```

Comment:
Please don't enable VTS in WindowsConsole.cs. We are going to enable it in `ANSIConsole` and try to keep `WindowsConsole` as old-school as possilbe (perhaps ONLY supporting 16 colors)...

Reply:
I need that for render the non-bmp properly. Otherwise it appear as bellow (see the black rectangle at right). This is only for output and not for input. If we want old-school work better I think we should keep it.

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/22601861-d765-425b-b49f-7503cf061e95)

Reply:
Only to remind you that I'm not using Ansi escape sequences on the old-school because it isn't really possible without using Windows Terminal. The only way I could find to resolve this was print each row and column with some control on the last column avoid jumping to the next row if a non-bmp is handled. To achieve this was only possible with the `ConsoleModes.EnableVirtualTerminalProcessing | ConsoleModes.DisableNewlineAutoReturn` together.

Reply:
I already removed `ConsoleModes.EnableVirtualTerminalProcessing`.

Reply:
Run a release version from cmd.exe (a new line is inserted at end):

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/3003b05d-857d-4187-bd10-b8648c174911)

Run a debug version from cmd.exe:

![image](https://github.com/gui-cs/Terminal.Gui/assets/13117724/2a1e8140-3d36-47d0-95d0-508dce6dec4b)

I really need to use `ConsoleModes.EnableVirtualTerminalProcessing | ConsoleModes.DisableNewlineAutoReturn` together for this working on release version as well. It seems that the VS2022 is dealing well with the `VSDebugger` even without `ConsoleModes.EnableVirtualTerminalProcessing`, but not with `cmd` or `conhost`. I only need it for dealing with endlines based on the type of the rune.